test/Microsoft.Extensions.DependencyInjection.Tests/ServiceCollectionServiceExtensionsTest.cs

```diff
@@ -395,5 +395,23 @@ public void Replace_ReplacesFirstServiceWithMatchingServiceType()
             // Assert
             Assert.Equal(new[] { descriptor2, descriptor3 }, collection);
         }
+
+        [Fact]
+        public void RemoveAll_RemovesAllServicesWithMatchingServiceType()
+        {
+            // Arrange
+            var collection = new ServiceCollection
+            {
+                new ServiceDescriptor(typeof(IFakeService), typeof(FakeService), ServiceLifetime.Transient),
+                new ServiceDescriptor(typeof(IFakeService), typeof(FakeService), ServiceLifetime.Transient),
+                new ServiceDescriptor(typeof(IFakeService), typeof(FakeService), ServiceLifetime.Transient)
+            };
+
+            // Act
+            collection.RemoveAll<IFakeService>();
```

Comment:
Can you keep some times in the list (so we know it's not just removing everything)?